internal/gatewayapi/runner/runner.go

```diff
@@ -61,15 +61,17 @@ func (r *Runner) subscribeAndTranslate(ctx context.Context) {
 		gatewayClasses := r.ProviderResources.GetGatewayClasses()
 		// Fetch the first gateway class since there should be only 1
 		// gateway class linked to this controller
-		t := &gatewayapi.Translator{
-			GatewayClassName: v1beta1.ObjectName(gatewayClasses[0].GetName()),
+		if gatewayClasses[0] != nil {
```

Comment:
you probably also want to check if `gatewayClasses` is nil or not to handle the case when there are no more gateway classes left / the subscribe notification was triggered due to a delete